presto-delta/src/test/java/com/facebook/presto/delta/TestTimestampColumns.java

```diff
@@ -0,0 +1,103 @@
+    }
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createDeltaQueryRunner(ImmutableMap.of(
+                "experimental.pushdown-subfields-enabled", "true",
+                "experimental.pushdown-dereference-enabled", "true"));
+    }
+
+    private static DistributedQueryRunner createDeltaQueryRunner(Map<String, String> extraProperties)
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog(DELTA_CATALOG)
+                .setSchema(DELTA_SCHEMA.toLowerCase(US))
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey("Europe/Madrid"))
+                .build();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
+                .setExtraProperties(extraProperties)
+                .build();
+
+        // Install the TPCH plugin for test data (not in Delta format)
+        queryRunner.installPlugin(new TpchPlugin());
+        queryRunner.createCatalog("tpch", "tpch");
+
+        Path dataDirectory = queryRunner.getCoordinator().getDataDirectory().resolve("delta_metadata");
+        Path catalogDirectory = dataDirectory.getParent().resolve("catalog");
+
+        // Install a Delta connector catalog
+        queryRunner.installPlugin(new DeltaPlugin());
+        Map<String, String> deltaProperties = ImmutableMap.<String, String>builder()
+                .put("hive.metastore", "file")
+                .put("hive.metastore.catalog.dir", catalogDirectory.toFile().toURI().toString())
+                .build();
+        queryRunner.createCatalog(DELTA_CATALOG, "delta", deltaProperties);
+
+        // Install a Hive connector catalog that uses the same metastore as Delta
+        // This catalog will be used to create tables in metastore as the Delta connector doesn't
+        // support creating tables yet.
+        queryRunner.installPlugin(new HivePlugin("hive"));
+        Map<String, String> hiveProperties = ImmutableMap.<String, String>builder()
+                .put("hive.metastore", "file")
+                .put("hive.metastore.catalog.dir", catalogDirectory.toFile().toURI().toString())
+                .put("hive.allow-drop-table", "true")
+                .put("hive.security", "legacy")
+                .build();
+        queryRunner.createCatalog(HIVE_CATALOG, "hive", hiveProperties);
+        queryRunner.execute(format("CREATE SCHEMA %s.%s", HIVE_CATALOG, DELTA_SCHEMA));
+
+        return queryRunner;
+    }
+
+    @Test(dataProvider = "deltaReaderVersions")
+    public void testDeltaTimezoneTypeSupport(String version)
+    {
+        String testQuery = format("select * from \"%s\".\"%s\"",
+                PATH_SCHEMA, goldenTablePathWithPrefix(version, "test-typing"));
+        MaterializedResult expectedResult = getQueryRunner().execute(testQuery);
+        assertEquals(expectedResult.getMaterializedRows().size(), 10);
+    }
```

Comment:
Can't we add this test in the existing https://github.com/prestodb/presto/blob/master/presto-delta/src/test/java/com/facebook/presto/delta/TestDeltaIntegration.java?